git-hub

```diff
@@ -816,61 +813,113 @@ class CloneCmd (object):
         parser.add_argument('--no-triangular', action="store_false",
                 dest='triangular',
                 help="do not use Git 'triangular workflow' setup")
-        return True # we need to get unknown arguments
 
     @classmethod
     def run(cls, parser, args):
-        (urltype, proj) = cls.parse_repo(args.repository)
-        (repo, upstream, forked) = cls.setup_repo(proj)
-        dest = args.dest or repo['name']
+        if args.repository is not None:
+            (urltype, proj) = cls.parse_repo(args.repository)
+        else:
+            git_remotes = git('remote', 'show', '-n').split('\n')
+            remote = args.upstreamremote
+            if remote not in git_remotes:
+                die("No REPO specified, nor does `{}` remote exist", remote)
```

Comment:
```suggestion
               parser.error("No REPO specified, nor does `{}` remote exist".format(remote))
```

This will use the parse to report the usage too.